src/semaphore/broadcast/markdown.py

```diff
@@ -142,6 +142,20 @@ def is_relevant_to_env(self, env_name: str) -> bool:
         else:
             return False
 
+    def summarize(
+        self, summary: Optional[str], body: Optional[str]
+    ) -> tuple[Optional[str], Optional[str]]:
+        if body is not None and summary is None:
+            paragraphs = body.split("\n\n")
+            new_summary = paragraphs[0]
+
+            del paragraphs[0]
+            new_body = "\n\n".join(paragraphs)
+            return new_summary, new_body
+
+        else:
+            return summary, body
```

Comment:
Here's how to hint to mypy that summary will never be `None`:

```suggestion
        else:
            if self.metadata.summary is None:
                raise RuntimeError("summary metadata must be set if body is empty")
            return summary, body
```